### PR TITLE
View controller: Send layout traits for component appear/disappear

### PR DESCRIPTION
**Pull request by JohnSundell**
This change makes `HUBViewController` send a component’s layout traits to its delegate as part of the methods called whenever a component is either about to appear, or just disappeared from the screen.

The reason for this change is that we want to enable API users to make decisions based on the layout traits, instead of resorting to comparing view frames (which are bound to change during the layout cycle).

### CODE REVIEW

**Comment by JohnSundell** [comment]
@spotify/objc-dev

File: include/HubFramework/HUBViewController.h [hunk 1]
```diff
@@ -86,21 +87,25 @@ NS_ASSUME_NONNULL_BEGIN
  *
  *  @param viewController The view controller in which a component is about to appear
  *  @param componentModel The model of the component that is about to appear
+ *  @param layoutTraits The layou traits of the component that is about to appear
```
Comment (krris) [inline include/HubFramework/HUBViewController.h]: Missing `t` in  `layou` 😄

File: include/HubFramework/HUBViewController.h [hunk 2]
```diff
@@ -86,21 +87,25 @@ NS_ASSUME_NONNULL_BEGIN
...
     componentWithModel:(id<HUBComponentModel>)componentModel
+          layoutTraits:(NSSet<HUBComponentLayoutTrait> *)layoutTraits
       willAppearInView:(UIView *)componentView;
 
 /**
  *  Sent to a Hub Framework view controller's delegate when a component disappeared from the screen
  *
  *  @param viewController The view controller in which a component disappeared
  *  @param componentModel The model of the component that disappeared
+ *  @param layoutTraits The layou traits of the component that disappeared
```
Comment (krris) [inline include/HubFramework/HUBViewController.h]: Same here 😉

File: tests/HUBViewControllerTests.m [hunk 3]
```diff
@@ -1126,17 +1131,21 @@ - (void)testComponentNotifiedOfViewWillAppearWhenCellIsDisplayed
     [collectionViewDelegate collectionView:self.collectionView willDisplayCell:cell forItemAtIndexPath:indexPath];
     [collectionViewDelegate collectionView:self.collectionView willDisplayCell:cell forItemAtIndexPath:indexPath];
     
-    XCTAssertEqual(self.component.numberOfAppearances, (NSUInteger)1);
-    XCTAssertEqual(self.componentModelsFromAppearanceDelegateMethod.count, (NSUInteger)1);
+    XCTAssertEqual(self.component.numberOfAppearances, 1u);
```
Comment (krris) [inline tests/HUBViewControllerTests.m]: Oh, I didn't know about this: `1u` == `(NSUInteger)1`

Comment (JohnSundell) [inline tests/HUBViewControllerTests.m]: @cerihughes taught me about it! So awesome 😄

**Comment by 8W9aG** [comment]
👍

**Pull request merged**